src/werkzeug/debug/tbtools.py

```diff
@@ -245,12 +245,14 @@ def __init__(self, exc_type, exc_value, tb):
         self.exception_type = exception_type
 
         self.groups = []
+        memo = set()
         while True:
             self.groups.append(Group(exc_type, exc_value, tb))
+            memo.add(exc_value)
```

Comment:
this assumes hashability of exceptions (which isn't always true), I wrote a fix for this: https://github.com/pallets/werkzeug/pull/1542